supabase/sql/5-llm-queries-and-responses.sql

```diff
@@ -0,0 +1,115 @@
+-- Create llm_queries table
+CREATE TABLE IF NOT EXISTS public.llm_queries (
+    query_id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
+    requester_id uuid NOT NULL,
+    prompt text NOT NULL,
+    model text NOT NULL,
+    response_types text[] NOT NULL,
+    hive_size integer NOT NULL,
+    perspective text NOT NULL,
+    age_range int[] NOT NULL,
+    income_range int[] NOT NULL,
+    status text DEFAULT 'pending',
+    created_at timestamp with time zone DEFAULT timezone('utc'::text, now()) NOT NULL
+);
+
+ALTER TABLE public.llm_queries ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "Users can view their own queries"
+ON public.llm_queries FOR SELECT
+TO authenticated
+USING (requester_id = auth.uid());
+
+CREATE POLICY "Users can insert their own queries"
+ON public.llm_queries FOR INSERT
+TO authenticated
+WITH CHECK (requester_id = auth.uid());
+
+
+-- Create llm_responses table
+CREATE TABLE IF NOT EXISTS public.llm_responses (
+    response_id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
+    query_id uuid NOT NULL REFERENCES public.llm_queries(query_id),
+    question text NOT NULL,
+    responses jsonb NOT NULL,
+    created_at timestamp with time zone DEFAULT timezone('utc'::text, now()) NOT NULL
+);
+
+ALTER TABLE public.llm_responses ENABLE ROW LEVEL SECURITY;
+
+CREATE POLICY "Users can view responses of their own queries"
+ON public.llm_responses FOR SELECT
+TO authenticated
+USING (
+  query_id IN (
+    SELECT q.query_id FROM public.llm_queries q WHERE q.requester_id = auth.uid()
+  )
+);
+
+-- Trigger function to handle new llm_query
+CREATE OR REPLACE FUNCTION public.handle_new_llm_query()
+RETURNS trigger AS $$
+DECLARE
+    service_role_key TEXT;
+    pgnet_id INTEGER;
+    http_status INTEGER;
+    required_credits integer := 1; -- Adjust credit cost as needed
+BEGIN
+    -- Check credits
+    IF NOT public.check_sufficient_credits(NEW.requester_id, required_credits) THEN
+        UPDATE public.llm_queries SET status = 'insufficient credits' WHERE query_id = NEW.query_id;
+        RETURN NEW;
+    END IF;
+
+    -- Consume credits
+    IF NOT public.consume_credits(NEW.requester_id, required_credits) THEN
+        UPDATE public.llm_queries SET status = 'insufficient credits' WHERE query_id = NEW.query_id;
+        RETURN NEW;
+    END IF;
+
+    UPDATE public.llm_queries SET status = 'processing' WHERE query_id = NEW.query_id;
+
+    SELECT decrypted_secret
+    INTO service_role_key
+    FROM vault.decrypted_secrets
+    WHERE name = 'SUPABASE_SERVICE_ROLE_KEY';
+
+    SELECT INTO pgnet_id net.http_post(
+        url := 'https://<your-supabase-project>.supabase.co/functions/v1/process-llm-query'::text,
```

Comment:
Our project is `ueguuautcrdolqpyyrjw`. It's usually not ideal to share this, but we don't have any other choice with a public repo. We just have to dot our i's and cross our t's elsewhere.